spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -384,4 +436,28 @@ public void acknowledge() {
 
 	}
 
+	private class MessageListener implements IMqttMessageListener {
+
+		MessageListener() {
+		}
+
+		@Override
+		public void messageArrived(String topic, MqttMessage message) {
+			Mqttv5PahoMessageDrivenChannelAdapter.this.messageArrived(topic, message);
```

Comment:
Also I wonder if this internal class makes sense, too, since we simply can use a lambda or better method reference: `IMqttMessageListener listener = this::messageArrived;`